core/unit_test/TestViewOutOfBoundsAccess.hpp

```diff
@@ -0,0 +1,171 @@
+  {
+    char buffer[64] = "my prefix ";
+    append_formated_multidimensional_index(buffer, 1);
+    EXPECT_STREQ(buffer, "my prefix [1]");
+  }
+  {
+    char buffer[64] = "I was here";
+    append_formated_multidimensional_index(buffer, 1, 2, 3);
+    EXPECT_STREQ(buffer, "I was here[1,2,3]");
+  }
+}
+
+template <class View, class ExecutionSpace>
+struct TestViewOutOfBoundAccess {
+  View v;
+  static constexpr auto rank = View::rank;
+
+  template <std::size_t... Is>
+  KOKKOS_FUNCTION decltype(auto) bad_access(std::index_sequence<Is...>) const {
+    return v((Is * 1 + Is == 0 ? v.extent(Is) + 3 : 0)...);
+  }
+
+  KOKKOS_FUNCTION void operator()(int) const {
+    ++bad_access(std::make_index_sequence<rank>{});
+  }
+
+  template <std::size_t... Is>
+  std::string get_details(std::index_sequence<Is...>) {
+    std::stringstream ss;
+    ss << "with indices \\[";
+    ((ss << (Is == 0 ? v.extent(Is) + 3 : 0)
+         << (Is == View::rank() - 1 ? "\\]" : ",")),
+     ...);
+    ss << " but extents \\[";
+    ((ss << v.extent(Is) << (Is == View::rank() - 1 ? "\\]" : ",")), ...);
+    return ss.str();
+  }
+
+  auto get_details() {
+    return get_details(std::make_index_sequence<View::rank()>());
+  }
+
+  TestViewOutOfBoundAccess(View w, ExecutionSpace const& s, std::string matcher)
+      : v(std::move(w)) {
+    constexpr bool view_accessible_from_execution_space =
+        Kokkos::SpaceAccessibility<
+            /*AccessSpace=*/ExecutionSpace,
+            /*MemorySpace=*/typename View::memory_space>::accessible;
+    EXPECT_TRUE(view_accessible_from_execution_space);
+
+    matcher += ".*" + get_details();
+
+    EXPECT_DEATH(
+        {
+          Kokkos::parallel_for(Kokkos::RangePolicy<ExecutionSpace>(s, 0, 1),
+                               *this);
+          Kokkos::fence();
+        },
+        matcher);
+  }
+};
+
+template <class View, class LblOrPtr, std::size_t... Is>
+auto make_view_impl(LblOrPtr x, std::index_sequence<Is...>) {
+  return View(x, (Is + 1)...);
+}
+
+template <class View, class LblOrPtr>
+auto make_view(LblOrPtr x) {
+  return make_view_impl<View>(std::move(x),
+                              std::make_index_sequence<View::rank>());
+}
+
+template <class ExecutionSpace>
+void test_view_out_of_bounds_access() {
+  ExecutionSpace const exec_space{};
+  // clang-format off
+  using V1 = Kokkos::View<int*,        ExecutionSpace>;
+  using V2 = Kokkos::View<int**,       ExecutionSpace>;
+  using V3 = Kokkos::View<int***,      ExecutionSpace>;
+  using V4 = Kokkos::View<int****,     ExecutionSpace>;
+  using V5 = Kokkos::View<int*****,    ExecutionSpace>;
+  using V6 = Kokkos::View<int******,   ExecutionSpace>;
+  using V7 = Kokkos::View<int*******,  ExecutionSpace>;
+  using V8 = Kokkos::View<int********, ExecutionSpace>;
+  constexpr bool is_device =
+      std::is_same_v<ExecutionSpace, Kokkos::DefaultExecutionSpace> &&
+      std::is_same_v<Kokkos::DefaultExecutionSpace,
```

Comment:
I had gotten it wrong anyway it should have been and default exec is NOT the same as default host exec